archive/app/controllers/ArchiveController.scala

```diff
@@ -30,33 +33,26 @@ class ArchiveController(redirects: RedirectService) extends Controller with Logg
 
   def lookup(path: String) = Action.async{ implicit request =>
 
-    val cachedArchiveRedirect = Cached(CacheTime.ArchiveRedirect) _
-
-    // lookup the path to see if we have a location for it in the database
-    lookupPath(path).map(_.map(cachedArchiveRedirect).getOrElse {
-
-      // if we do not have a location in the database then follow these rules
-      path match {
-        case Gallery(gallery)      => redirectTo(gallery)
-        case Century(century)      => redirectTo(century)
-        case Guardian(endOfUrl)    => redirectTo(endOfUrl)
-        case Lowercase(lower)      => redirectTo(lower)
-
-        // Googlebot hits a bunch of really old combiners and combiner RSS
-        // bounce these to the section
-        case CombinerSectionRss(section)                => redirectTo(s"$section/rss")
-        case CombinerSection(section)                   => redirectTo(section)
-        case Combiner(combiner)                         => redirectTo(combiner)
-        case DatedSpecialIndexPage(section, rest, _)    => redirectTo(section, rest, "all")
-        case SectionSpecialIndex(section, _)            => redirectTo(section, "all")
-        case NewspaperPage(paper, date, book)           => redirectTo(paper, book, date, "all")
-
-        case _ =>
+    lookupPath(path)
+      .map { _
+        .map(Cached(CacheTime.ArchiveRedirect))
+        .orElse(redirectForPath(path))
+      }
+      .flatMap { _
+        .map(Future.successful)
+        .getOrElse {
           log404(request)
-          // short cache time as we might just be waiting for the content api to index
-          Cached(CacheTime.NotFound)(WithoutRevalidationResult(NotFound(views.html.notFound())))
+          def notFoundResponse(html: Html): Result = Cached(CacheTime.NotFound)(WithoutRevalidationResult(NotFound(html)))
+          if (ABJavascriptRenderingVariant.isParticipating) {
+            renderer
+              .render(ui.NotFound)
+              .map(notFoundResponse)
+          } else {
+            Future.successful(notFoundResponse(views.html.notFound()))
+          }
+        }
       }
-    })
+
```

Comment:
I am not a huge fan of this implementation. Shout out if you can think of a better way to achieve the same thing